cirq/vis/histogram.py

```diff
@@ -0,0 +1,150 @@
+    ax: Optional[plt.Axes] = None,
+    *,
+    cdf_on_x: bool = False,
+    axis_label: str = '',
+    semilog: bool = True,
+    median_line: bool = True,
+    median_label: Optional[str] = 'median',
+    mean_line: bool = False,
+    mean_label: Optional[str] = 'mean',
+    hide_zero: bool = True,
+    title: Optional[str] = None,
+    **kwargs,
+) -> plt.Axes:
+    """Plot the integrated histogram for an array of data.
+
+    Suppose the input is a list of gate fidelities. The x-axis of the plot will
+    be gate fidelity, and the y-axis will be the probability that a random gate
+    fidelity from the list is less than the x-value. It will look something like
+    this
+
+    1.0
+    |              |
+    |           ___|
+    |           |
+    |       ____|
+    |      |
+    |      |
+    |_____|_______________
+    0.0
+
+    Another way of saying this is that we assume the probability distribution
+    function (pdf) of gate fidelities is a set of equally weighted delta
+    functions at each value in the list. Then, the "integrated histogram"
+    is the cumulative distribution function (cdf) for this pdf.
+
+    Args:
+        data: Data to histogram. If the data is a mapping, we histogram the
+            values. All nans will be removed.
+        ax: The axis to plot on. If None, we generate one.
+        cdf_on_x: If True, flip the axes compared the above example.
+        axis_label: Label for x axis.
+        semilog: If True, force the x-axis to be logarithmic.
+        median_line: If True, draw a vertical line on the median value.
+        median_label: If drawing median line, optional label for it.
+        mean_line: If True, draw a vertical line on the mean value.
+        mean_label: If drawing mean line, optional label for it.
+        **plot_options: Kwargs to forward to `ax.step()`. Some examples are
+            color: Color of the line.
+            linestyle: Linestyle to use for the plot.
+            lw: linewidth for integrated histogram
+            ms: marker size for a histogram trace
+            label: An optional label which can be used in a legend.
+
+
+    Returns:
+        The axis that was plotted on.
+    """
+    show_plot = not ax
+    if ax is None:
+        fig, ax = plt.subplots(1, 1)
+
+    if isinstance(data, Mapping):
+        data = list(data.values())
+
+    data = [d for d in data if not np.isnan(d)]
+    n = len(data)
+
+    if not hide_zero:
+        bin_values = np.linspace(0, 1, n + 1)
+        parameter_values = sorted(np.concatenate(([0], data)))
+    else:
+        bin_values = np.linspace(0, 1, n)
+        parameter_values = sorted(data)
+    plot_options = {
+        "where": 'post',
+        "color": 'b',
+        "linestyle": '-',
+        "lw": 1.0,
+        "ms": 0.0,
+    }
+    plot_options.update(kwargs)
+    if cdf_on_x:
+        ax.step(bin_values, parameter_values, **plot_options)
+        setsemilog = ax.semilogy
```

Comment:
1. can you add underscores to these variable names? 
2. also, I think inline (e.g. `set_semilog = ax.semilogy if cdf_on_x else ax.semilogx`) would be a bit less error prone in this case - instead of maintaining two lists we'll have a single one.

Reply:
done.